src/components/List/List.jsx

```diff
@@ -43,18 +46,27 @@ const List = () => {
       )}
 
       {value && !value.empty && (
-        <ul>
-          {value.docs.map((doc) => (
-            <li key={doc.id}>
-              <input
-                type="checkbox"
-                checked={has24HoursPassed(doc.data().lastDate)}
-                onChange={() => handleCheck(doc.id, doc.data().lastDate)}
-              />
-              {doc.data().name}
-            </li>
-          ))}
-        </ul>
+        <>
+          <input value={filter} onChange={(e) => setFilter(e.target.value)} />
+          {filter.length >= 1 && (
+            <button onClick={() => setFilter('')}>X</button>
+          )}
+
+          <ul>
+            {value.docs
+              .filter((doc) => doc.data().name.includes(filter))
```

Comment:
I don't fully understand how the filter is working as .includes returns a boolean value and based on this value the filter is applied. Nice job.

Reply:
sounds like you understand it!

Reply:
> I don't fully understand how the filter is working as .includes returns a boolean value and based on this value the filter is applied. Nice job.

the filter method returns an array with all the elements that pass the condition, if includes returns true, then the element will be added to the array in filter, and since filter returns an array, we can map over it.